modules/renter/download.go

```diff
@@ -520,3 +524,53 @@ func (r *Renter) DownloadHistory() []modules.DownloadInfo {
 	}
 	return downloads
 }
+
+// ClearDownloadHistory clears the renter's download history based on
+// provided start and end times
+func (r *Renter) ClearDownloadHistory(start, end time.Time) error {
+	if err := r.tg.Add(); err != nil {
+		return err
+	}
+	defer r.tg.Done()
+	r.downloadHistoryMu.Lock()
+	defer r.downloadHistoryMu.Unlock()
+
+	// Clear download history if both start and end times are zero values
+	if start.Equal(time.Unix(0, 0)) && end.Equal(time.Unix(0, 0)) {
+		r.downloadHistory = r.downloadHistory[:0]
+		return nil
+	}
+
+	// If End is zero value, search for start time to clear history
+	if end.Equal(time.Unix(0, 0)) {
+		i := sort.Search(len(r.downloadHistory), func(i int) bool { return r.downloadHistory[i].staticStartTime.After(start) })
+		if i > len(r.downloadHistory) {
+			// Start time is After all downloads, clear entire download history
+			r.downloadHistory = r.downloadHistory[:0]
```

Comment:
if `start` is after all downloads, I would expect nothing to be cleared

Reply:
yea that is a typo, it originally said "end is after" so it should be "start is before"